packages/components/src/autocomplete/index.tsx

```diff
@@ -163,19 +194,35 @@ export function useAutocomplete( {
 		) {
 			return;
 		}
+
+		let newIndex = 0;
```

Comment:
Code style nit: a standard way to use local variables in a `case` is to wrap the `case` clause in a block:
```js
case 'ArrowUp': {
  const newIndex = ...
  ...
}
```

Reply:
Done.